src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -1298,6 +1299,7 @@ private static void cleanUpLoader(ClassLoader loader, Set<ClassLoader> encounter
             if (encounteredClasses.add(clazz)) {
                 LOGGER.log(Level.FINER, "found {0}", clazz.getName());
                 Introspector.flushFromCaches(clazz);
+                cleanUpClassInfoCache(clazz);
```

Comment:
Any idea if the same issue applies to sandboxed non-Pipeline Groovy scripts as well? This code duplicates https://github.com/jenkinsci/script-security-plugin/blob/35f6a0b8207ed3a32a85f27c1312da6cd738eeaa/src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java#L205-L214.